internal/translation/dbuser/conversion.go

```diff
@@ -0,0 +1,230 @@
+)
+
+type User struct {
+	*akov2.AtlasDatabaseUserSpec
+	Password  string
+	ProjectID string
+}
+
+// NewUser wraps a Kubernetes Atlas User Spec pointer augmenting it with projectID and password.
+func NewUser(spec *akov2.AtlasDatabaseUserSpec, projectID, password string) (*User, error) {
+	return normalize(&User{AtlasDatabaseUserSpec: spec, ProjectID: projectID, Password: password})
+}
+
+// DiffSpecs returns all differences found in the user Spec fields or a spec user and an atlas user.
+// Non Spec fields are not compared. Inputs are dbuser.User so they are normalized
+func DiffSpecs(specUser, atlasUser *User) []string {
+	diffs := []string{}
+	if atlasUser.Username != specUser.Username {
+		diffs = append(diffs, fmt.Sprintf("Usernames differs from spec: %q <> %q\n",
+			atlasUser.Username, specUser.Username))
+	}
+	if atlasUser.DatabaseName != specUser.DatabaseName {
+		diffs = append(diffs, fmt.Sprintf("DatabaseName differs from spec: %q <> %q\n",
+			atlasUser.DatabaseName, specUser.DatabaseName))
+	}
+	if atlasUser.DeleteAfterDate != specUser.DeleteAfterDate {
+		diffs = append(diffs, fmt.Sprintf("DeleteAfterDate differs from spec: %q <> %q\n",
+			atlasUser.DeleteAfterDate, specUser.DeleteAfterDate))
+	}
+	if atlasUser.OIDCAuthType != specUser.OIDCAuthType {
+		diffs = append(diffs, fmt.Sprintf("OIDCAuthType differs from spec: %q <> %q\n",
+			atlasUser.OIDCAuthType, specUser.OIDCAuthType))
+	}
+	if atlasUser.AWSIAMType != specUser.AWSIAMType {
+		diffs = append(diffs, fmt.Sprintf("AWSIAMType differs from spec: %q <> %q\n",
+			atlasUser.AWSIAMType, specUser.AWSIAMType))
+	}
+	if atlasUser.X509Type != specUser.X509Type {
+		diffs = append(diffs, fmt.Sprintf("X509Type differs from spec: %q <> %q\n",
+			atlasUser.X509Type, specUser.X509Type))
+	}
+	if !reflect.DeepEqual(atlasUser.Roles, specUser.Roles) {
+		diffs = append(diffs, fmt.Sprintf("Roles differs from spec: %v <> %v\n",
+			atlasUser.Roles, specUser.Roles))
+	}
+	if !reflect.DeepEqual(atlasUser.Scopes, specUser.Scopes) {
+		diffs = append(diffs, fmt.Sprintf("Scopes differs from spec: %v <> %v END\n",
+			atlasUser.Scopes, specUser.Scopes))
+	}
+	return diffs
+}
+
+func normalize(user *User) (*User, error) {
+	cmp.NormalizeSlice(user.Roles, func(i, j int) bool {
+		return user.Roles[i].RoleName < user.Roles[j].RoleName &&
+			user.Roles[i].DatabaseName < user.Roles[j].DatabaseName &&
+			user.Roles[i].CollectionName < user.Roles[j].CollectionName
+	})
+	cmp.NormalizeSlice(user.Scopes, func(i, j int) bool {
+		return user.Scopes[i].Name < user.Scopes[j].Name &&
+			user.Scopes[i].Type < user.Scopes[j].Type
+	})
+	if user.DeleteAfterDate != "" { // enforce date format
+		operatorDeleteDate, err := timeutil.ParseISO8601(user.DeleteAfterDate)
+		if err != nil {
+			return nil, err
+		}
+		user.DeleteAfterDate = timeutil.FormatISO8601(operatorDeleteDate)
+	}
+	// Ensure comparisons succeed on default value
+	// TODO: Check if this is required after updating the Helm Chart CRDs
```

Comment:
same as below: please create a ticket and attach `TODO(CLOUDP-....`)

Reply:
I can act on this now, the release is done, so the new helm chart already takes this into account. Should be able to remove.

Reply:
next update will test this removal